tests/test_model_manager.py

```diff
@@ -36,156 +35,278 @@
 from unittest.mock import patch
 
 
+class MockRunConfigs():
+
+    def __init__(self):
+        self._configs = []
+
+    def get_configs(self):
+        return set(self._configs)
+
+    def add_config(self, config):
+        """ 
+        Add a single config from a dict
+        """
+        list_of_tuples = [
+            (y, config[y]) for y in sorted(config.keys()) if config[y] != None
+        ]
+        out_tuple = tuple(item for x in list_of_tuples for item in x)
```

Comment:
(nit) This feels like one of those things you can "technically" do but its unclear that you should. It is hard to read list comprehensions as they get more complex. I would prefer we wrote this a different way. Would this provide the functionality you need: https://docs.python.org/3/library/functions.html#zip

Reply:
Zip won't work. I kind of need the opposite of zip. I also need it to be explicitly sorted and a tuple so it can be hashed as a set and easily compared.  I have improved the documentation.